### DISCUSSION TITLE
How to link storage into table

### DISCUSSION THREAD

**Discussion by nicogulo** [supabase/supabase#14027] [Questions]
Hi, I just new here. there is any documentation for link storage into table? I just try to add relation into storage like this

![image](https://user-images.githubusercontent.com/63594315/235292750-af2b30dc-4e2d-4806-85c1-f646498dd915.png)

after that, I insert `image_name` with name bucket (`app`)

![image](https://user-images.githubusercontent.com/63594315/235292825-7ebaf09d-6cb0-4d52-affb-dd72e0daa5af.png)

in that mean, I want to use file inside folder in `app`, but I don't know how.

![image](https://user-images.githubusercontent.com/63594315/235293058-00c45526-b513-44df-8074-471cf40a8316.png)

can you guys help me? Thanks a Lot

**Comment by GaryAustin1** [comment]
Not quite clear on your goal. You really only need to store pathname (and bucket name if not fixed for the class of data) in database tables. You have to make a storage API call to do any operation on storage. For instance you can't just cascade delete a row in storage.objects as that does not delete the file data from s3 storage and just orphans it. So linking a storage object as a fk just means you have to go fetch the pathname and bucket name anyway to carry out any sort of operation.

Also DO NOT modify Supabase schemas except possibly for trigger functions. Supabase can change the columns and reset them at any point.
